documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,632 @@
+* The Kafka cluster configured to connect with Keycloak is deployed to your Kubernetes cluster.
+
+.Procedure
+. In the Keycloak Admin Console, go to *Clients* > *kafka* > *Authorization* > *Settings* and check that *Decision Strategy* is set to *Affirmative*.
++
+An affirmative policy means that at least one policy must be satisfied for a client to access the Kafka cluster.
+
+. Run a new interactive pod container using the Strimzi Kafka image to connect to a running Kafka broker.
++
+[source,shell,subs="attributes"]
+----
+NS=sso
+kubectl run -ti --restart=Never --image={DockerKafkaImageCurrent} kafka-cli -n $NS -- /bin/sh
+----
++
+NOTE: If `kubectl` times out waiting on the image download, subsequent attempts may result in an _AlreadyExists_ error.
+
+. Attach to the pod container.
++
+[source,shell]
+----
+kubectl attach -ti kafka-cli -n $NS
+----
+
+. Use the hostname of the Keycloak instance to prepare a certificate for client connection using TLS.
++
+[source,shell,subs="+quotes"]
+----
+SSO_HOST=_SSO-HOSTNAME_
+SSO_HOST_PORT=$SSO_HOST:443
+STOREPASS=storepass
+
+echo "Q" | openssl s_client -showcerts -connect $SSO_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/sso.crt
+----
+
+. Create a truststore for TLS connection to the Kafka brokers.
++
+[source,shell,subs="+quotes"]
+----
+keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias sso -storepass $STOREPASS -import -file /tmp/sso.crt -noprompt
+----
+
+. Use the Kafka bootstrap address as the hostname of the Kafka broker and the `tls` listener port (9093) to prepare a certificate for the Kafka broker.
++
+[source,shell]
+----
+KAFKA_HOST_PORT=my-cluster-kafka-bootstrap:9093
+STOREPASS=storepass
+
+echo "Q" | openssl s_client -showcerts -connect $KAFKA_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/my-cluster-kafka.crt
+----
+
+. Add the certificate for the Kafka broker to the truststore.
++
+[source,shell]
+----
+keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias my-cluster-kafka -storepass $STOREPASS -import -file /tmp/my-cluster-kafka.crt -noprompt
+----
+
+[id='proc-oauth-authorization-keycloak-example-check_{context}']
+== Checking authorized access to Kafka using permissions
+
+Check the authorization rules applied through the Keycloak realm.
+Apply the checks using Kafka's example producer and consumer clients to create topics with user and service accounts that have different levels of access.
+
+Use the `team-a-client` and `team-b-client` clients to check the authorization rules.
+Use the `alice` admin user to perform additional administrative tasks on Kafka.
+
+.Prerequisites
+
+* ZooKeeper and Kafka are running in the Kubernetes cluster to be able to send and receive messages.
+* Access between the Kafka brokers and Keycloak is authenticated.
+* The latest Kafka producer and consumer binaries are installed on your local machine.
```

Comment:
In this example we're using a specific released version of Strimzi Kafka image which already contains the latest binaries. 
```suggestion
```